src/module.h

```diff
@@ -43,6 +43,34 @@ do {                                            \
 #define RedisModule_ReplyWithLiteral(ctx, literal) \
   RedisModule_ReplyWithStringBuffer(ctx, literal, sizeof(literal) - 1)
 
+// A macro that extracts the first argument from a variadic macro
+#define SECOND_ARG_HELPER(first, second, ...) second
+#define SECOND_ARG(...) SECOND_ARG_HELPER(__VA_ARGS__)
+
+#define RM_CREATE_COMMAND(aclCategories, ...)                                  \
```

Comment:
instead of a macro can this be a function?
Then the acl param won't have to be first.

Reply:
Why does it bother that it's first?
It obviously can be a function, but I don't see a huge benefit here

Reply:
Mainly maintainability.
It is much easier to maintain a function than a macro.
Also consider you stopped using RM_TRY which means we won't get verbose prints if the function succeeds.

For example if someone introduces a code which returns early from the macro then it would prevent all the following functions from being called.

Reply:
This would be a big change that I don't think is necessary. I went with the currently implemented macro approach that overall makes sense here and is comfortable. If someone would need to return early, he would need it to do it appropriately, sure.
Regarding the log, you practically don't see it today anyways, and there is no need for this positive log IMO.

Reply:
If there was a failure - you know about it since the server fails. Otherwise - all is good. This is the way we log today and are supposed to.
Why don't we also report more successes in our code-base? It garbages the log for no reason

Reply:
Will consider this since it's more comfortable at the end of the day..